api/src/test/java/org/pmiops/workbench/api/BigQueryBaseTest.java

```diff
@@ -0,0 +1,107 @@
+package org.pmiops.workbench.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.bigquery.*;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.junit.After;
+import org.junit.Before;
+import org.pmiops.workbench.api.config.TestBigQueryConfig;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.lang.reflect.Type;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+@SpringBootTest(classes = {TestBigQueryConfig.class})
+public abstract class BigQueryBaseTest {
+
+    private static final Logger log = Logger.getLogger(BigQueryBaseTest.class.getName());
+
+    @Autowired
+    BigQuery bigquery;
+
+    @Autowired
+    WorkbenchConfig workbenchConfig;
+
+    public static final String BASE_PATH = "src/test/resources/bigquery/";
+
+    @Before
+    public void setUp() throws Exception {
+        createDataSet(workbenchConfig.bigquery.dataSetId);
+        for (String tableName: getTableNames()) {
+            createTable(workbenchConfig.bigquery.dataSetId, tableName);
+            insertData(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+    }
+
+    @After
+    public void tearDown() {
+        for (String tableName: getTableNames()) {
+            deleteTable(workbenchConfig.bigquery.dataSetId, tableName);
+        }
+        deleteDataSet(workbenchConfig.bigquery.dataSetId);
```

Comment:
One thing of note is: if a test dies while executing, data sets may not get cleaned up.

That's probably fine... may be worth adding a comment that we may need to periodically clean up orphaned data sets here.

Reply:
Yes, this is always a possibility.